woodwork/logical_types.py

```diff
@@ -820,6 +825,26 @@ def _regex_validate(regex_key, series, return_invalid_values):
             raise TypeValidationError(info)
 
 
+def _replace_nans(series):
+    """
+    Replaces empty string values, string representations of NaN values ("nan", "<NA>"), and NaN equivalents
+    with np.nan or pd.NA depending on column dtype.
+    """
+    original_dtype = series.dtype
+    if str(original_dtype) == "string":
+        series = series.replace(ww.config.get_option("nan_values"), pd.NA)
+        return series
+    series = series.apply(
```

Comment:
Can we run some performance tests with and without this change? I have some concerns that it might lead to slower init in Woodwork

Reply:
I've run performance tests [here](https://alteryx.atlassian.net/wiki/spaces/PS/pages/1410727994/Performance+Tests+-+Expanding+nan+Inference+for+IntegerNullable+BooleanNullable). There's a significant increase in inference runtime when this line is added and I'm fine with removing it

Reply:
Yah I agree, let's remove casting, and let pandas handle it internally